src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -68,17 +66,16 @@ internal static HttpResponseMessage CreateResponse(HttpRequestMessage request, o
                 }
             }
 
-            // see if the content is a response object, defining http response
-            // properties
+            // see if the content is a response object, defining http response properties
             IDictionary<string, object> responseObject = null;
             if (content is JObject)
             {
-                responseObject = JsonConvert.DeserializeObject<Dictionary<string, object>>(stringContent, _dictionaryJsonConverter);
```

Comment:
The whole motivator for these changes was my realization that on this codepath, we could end up calling ToString on this IDictionary in CreateResultContent. So simplifying this so we're dealing with an ExpandoObject in both cases. One of the tests I added failed before these changes.